### PR TITLE
SPEC file: sync how we build snapshot and release packages

### PR DESCRIPTION
**Pull request by clebergnu**
Downstream (Fedora and EPEL) RPM packages have had a different way
of building release (by default) and snapshot packages by editing
the SPEC file.

Let's sync with them, while also allowing snapshot packages to be
built from our usual `make rpm` command.  Release builds are still
supported with `make rpm-release`.

Signed-off-by: Cleber Rosa <crosa@redhat.com>

### CODE REVIEW

**Comment by clebergnu** [comment]
[Backport Official Fedora SPEC files](https://trello.com/c/uWKfxZHu/887-backport-official-fedora-spec-files)

File: Makefile [hunk 1]
```diff
@@ -64,11 +65,11 @@ all:
 
 source: clean
 	if test ! -d SOURCES; then mkdir SOURCES; fi
-	git archive --prefix="avocado-$(COMMIT)/" -o "SOURCES/avocado-$(VERSION)-$(SHORT_COMMIT).tar.gz" HEAD
+	git archive --prefix="avocado-$(COMMIT)/" -o "SOURCES/avocado-$(SHORT_COMMIT).tar.gz" HEAD
 
 source-release: clean
 	if test ! -d SOURCES; then mkdir SOURCES; fi
-	git archive --prefix="avocado-$(RELEASE_COMMIT)/" -o "SOURCES/avocado-$(VERSION)-$(RELEASE_SHORT_COMMIT).tar.gz" $(VERSION)
```
Comment (ldoktor) [inline Makefile]: The `RELEASE_COMMIT` is not used anymore, you can remove it

File: python-avocado.spec [hunk 2]
```diff
@@ -1,8 +1,24 @@
 %global srcname avocado
+
+# Conditional for release vs. snapshot builds. Set to 1 for release build.
+%if ! 0%{?rel_build:1}
```
Comment (ldoktor) [inline python-avocado.spec]: Would it be possible to change this to `release_build`? The `rel_build` is quite a cryptic name... (not a strong requirement, though)

Comment (clebergnu) [inline python-avocado.spec]: As the goal here is to **sync** with what is already on downstream packages, the answer is no.

File: Makefile (hunk 1 above)
Comment (clebergnu) [inline Makefile]: I think it's still useful, both in the output of `make variables`, and as a way to show where the short commit comes from.

File: python-avocado.spec [hunk 3]
```diff
@@ -1,8 +1,24 @@
...
 %if ! 0%{?commit:1}
- %define commit 1d717446d28ff852fe5de895f14c7985c63ada3e
+%global commit		817c25c2bbedb6d43e5fa2ad0a78698f1416ab49
+%endif
+%if ! 0%{?commit_date:1}
+%global commit_date	20170407
+%endif
+%global shortcommit	%(c=%{commit};echo ${c:0:7})
+%global gitrel		.%{commit_date}git%{shortcommit}
+%global gittar		%{srcname}-%{shortcommit}.tar.gz
```
Comment (ldoktor) [inline python-avocado.spec]: This is really confusing, but after re-arranging it looks like it's correct... How about:

```diff
diff --git a/python-avocado.spec b/python-avocado.spec
index 82c49ee..78cb6bc 100644
--- a/python-avocado.spec
+++ b/python-avocado.spec
@@ -7,17 +7,17 @@
 
 # Settings used for build from snapshots.
 %if 0%{?rel_build}
-%global gittar         %{srcname}-%{version}.tar.gz
+       %global gittar          %{srcname}-%{version}.tar.gz
 %else
-%if ! 0%{?commit:1}
-%global commit         817c25c2bbedb6d43e5fa2ad0a78698f1416ab49
-%endif
-%if ! 0%{?commit_date:1}
-%global commit_date    20170407
-%endif
-%global shortcommit    %(c=%{commit};echo ${c:0:7})
-%global gitrel         .%{commit_date}git%{shortcommit}
-%global gittar         %{srcname}-%{shortcommit}.tar.gz
+       %if ! 0%{?commit:1}
+               %global commit          817c25c2bbedb6d43e5fa2ad0a78698f1416ab49
+       %endif
+       %if ! 0%{?commit_date:1}
+               %global commit_date     20170407
+       %endif
+       %global shortcommit     %(c=%{commit};echo ${c:0:7})
+       %global gitrel          .%{commit_date}git%{shortcommit}
+       %global gittar          %{srcname}-%{shortcommit}.tar.gz
 %endif
 
 # selftests are provided but may need to skipped because many of
```

Anyway if I understand this correctly (not an RPM expert) the main difference is that the released version will use the `tag` instead of `sha` as a pointer to the sources and some other name-related changes?

Comment (clebergnu) [inline python-avocado.spec]: Our `make rpm-release` target already used the tag reference, but it used to reference a source tarball that was named after the commit that matches the tag reference.  Now, it uses a version number for release builds **also** in the tarball.

With regards to style (indentation), please take into account:
1. It's not common to use indentation in a SPEC file
2. The goal is to **sync** with what's being done downstream, not to add "different differences"

Comment (ldoktor) [inline python-avocado.spec]: Sure, I was just asking, as I said I'm not a SPECfile expert, it just looked odd and without indentation it is really hard to follow. Anyway it seems to work well and the links look sane as well. So let's get this in and it'd be nice to suggest the `rel_build` variable name change to downstream as it's not really human friendly...

Comment (ldoktor) [inline python-avocado.spec]: Btw is it not? https://fedoraproject.org/wiki/PeterGordon/SpecFormattingGuidelines (the example is not indented, but the description says it should be. And googling I found several patches to use indentation to make visually clear the nested conditional blocks... Anyway let's sync the implementations and then we can try to improve it both, up and downstream....

Comment (clebergnu) [inline python-avocado.spec]: @ldoktor this is "someone"s note on **his own** guidelines, dated from 2008.  As you mentioned, not even his examples follow the suggestion. TBH, that is worth nothing to me.
